File: Programmers/연습문제/Level 1/숫자 문자열과 영단어/Solution.js
function solution(s) {
    const answer = eng2num(s);
    return answer;
}

function eng2num(str) {
    const dict = {
        'zero': 0,
        'one': 1,
        'two': 2,
        'three': 3,
        'four': 4,
        'five': 5,
        'six': 6,
        'seven': 7,
        'eight': 8,
        'nine': 9
    } 

    for (const eng in dict) {
        str = str.replace(new RegExp(eng, 'g'), dict[eng]);
    }

    return Number(str);
}

s = "one4seveneight";
console.log(solution(s));

File: Programmers/연습문제/Level 1/숫자 문자열과 영단어/Solution.swift
import Foundation

func solution(_ s:String) -> Int {
    let arr = ["zero", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine"]
    
    var result = s
    for a in arr.enumerated() {
        result = result.replacingOccurrences(of: a.element, with: String(a.offset))
    }
    return Int(result) ?? 0
}


// TC
let s = "one4seveneight"
print(solution(s)) // 1478
